spec/features/dashboard/collection_form_spec.rb

```diff
@@ -10,7 +10,20 @@
   let(:new_collection) { Collection.last }
 
   before do
+    mock_solr_indexing_job
+  end
+
+  # When stepping through each page of the form, we want to test whether the
+  # collection was indexed, and how, based on the buttons that were pressed.
+  # Because there can be multiple of these steps within each test, and because
+  # RSpec mocks _cumulatively_ record the number of times they've been called,
+  # we need a way to say "from this exact point, you should have been called
+  # once." We accomplish this by tearing down the mock and setting it back up.
+  def mock_solr_indexing_job
+    RSpec::Mocks.space.proxy_for(SolrIndexingJob)&.reset
```

Comment:
@rschenk OK, this makes sense. So basically any time this method is invoked, the count is reset and the expectation can be "this was called once" or "this thing was called at least once" or it's always going from zero to 1 or more.

Reply:
Exactly.

Otherwise, the factories can alter the count, as can previous pages in the form if testing the entire form end-to-end. This always resets it to 0 so you can be sure that your exact action is affecting the method call counts